pandas/core/internals/managers.py

```diff
@@ -926,22 +927,29 @@ def __init__(
                         f"Number of Block dimensions ({block.ndim}) must equal "
                         f"number of axes ({self.ndim})"
                     )
-                if isinstance(block, DatetimeTZBlock) and block.values.ndim == 1:
+                if block.values.ndim == 1 and (
+                    isinstance(block, DatetimeTZBlock) or block.dtype.type is Period
+                ):
                     # TODO(2.0): remove once fastparquet no longer needs this
-                    warnings.warn(
-                        "In a future version, the BlockManager constructor "
-                        "will assume that a DatetimeTZBlock with block.ndim==2 "
-                        "has block.values.ndim == 2.",
-                        DeprecationWarning,
-                        stacklevel=find_stack_level(),
-                    )
+                    if isinstance(block, DatetimeTZBlock):
+                        warnings.warn(
+                            "In a future version, the BlockManager constructor "
+                            "will assume that a DatetimeTZBlock with block.ndim==2 "
+                            "has block.values.ndim == 2.",
+                            DeprecationWarning,
+                            stacklevel=find_stack_level(),
+                        )
+                    else:
+                        warnings.warn(
```

Comment:
these warnings tested?

Reply:
not the new one; need to revert it since ive xfailed the cases that get here